charts/studio/templates/secret-studio.yaml

```diff
@@ -35,12 +37,13 @@ stringData:
   {{- end }}
 
   {{- if .Values.global.scmProviders.github.privateKey }}
-  GITHUB_APP_PRIVATE_KEY_PEM: {{ .Values.global.scmProviders.github.privateKey }}
+  GITHUB_APP_PRIVATE_KEY_PEM: |- {{ .Values.global.scmProviders.github.privateKey | nindent 4 }}
```

Comment:
This is where we have a bug now :(

Reply:
I am not seeing this bug,  I've set my values:
```yaml
global:
  scmProviders:
    github:
      ...
      privateKey: |-
        -----BEGIN RSA PRIVATE KEY-----
        ....
```
:)
To understand this bug can you share how did you override this value in your `studio_values.yaml` (or whatever you named it)?

Reply:
With this values:
```yaml
global:
  scmProviders:
    github:
      privateKey: |-
            -----BEGIN RSA PRIVATE KEY-----
            -----END RSA PRIVATE KEY-----
```
the template fails with error:
``` 
$  helm template . -s templates/secret-studio.yaml --debug
install.go:192: [debug] Original chart version: ""
install.go:209: [debug] CHART PATH: /Users/marcin/git/github.com/iterative/helm-charts/charts/studio

---
# Source: studio/templates/secret-studio.yaml
apiVersion: v1
kind: Secret
metadata:
  name: studio
  annotations:
    "helm.sh/resource-policy": "keep"
  labels:
    helm.sh/chart: studio-0.1.11
    app.kubernetes.io/name: studio
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/version: "v1.41.3"
    app.kubernetes.io/managed-by: Helm
type: Opaque
stringData:
  DATABASE_URL: "psql://postgres:postgres@studio-postgresql:5432/iterativeai"
  GITHUB_APP_CLIENT_ID: secret
  GITHUB_APP_ID: secret
  GITHUB_APP_SECRET_KEY: secret
  GITHUB_APP_PRIVATE_KEY_PEM: -----BEGIN RSA PRIVATE KEY-----
Error: YAML parse error on studio/templates/secret-studio.yaml: error unmarshaling JSON: while decoding JSON: json: cannot unmarshal string into Go value of type releaseutil.SimpleHead

```
**Why?**
Because the fist line inlined with `GITHUB_APP_PRIVATE_KEY_PEM` and second line starts from position 0

For this setting it is more visible
```yaml
          privateKey: |-
            abc
            def
```
Running template debug returns:
```bash
$ helm template . -s templates/secret-studio.yaml --debug
install.go:192: [debug] Original chart version: ""
install.go:209: [debug] CHART PATH: /Users/marcin/git/github.com/iterative/helm-charts/charts/studio

---
# Source: studio/templates/secret-studio.yaml
apiVersion: v1
kind: Secret
metadata:
  name: studio
  annotations:
    "helm.sh/resource-policy": "keep"
  labels:
    helm.sh/chart: studio-0.1.11
    app.kubernetes.io/name: studio
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/version: "v1.41.3"
    app.kubernetes.io/managed-by: Helm
type: Opaque
stringData:
  DATABASE_URL: "psql://postgres:postgres@studio-postgresql:5432/iterativeai"
  GITHUB_APP_CLIENT_ID: "secret"
  GITHUB_APP_ID: "secret"
  GITHUB_APP_SECRET_KEY: "secret"
  GITHUB_APP_PRIVATE_KEY_PEM: abc
def

  # Set secretKey to existing value or generate a random one
  SECRET_KEY: "secret"
  BLOBVAULT_AWS_ACCESS_KEY_ID: admin
  BLOBVAULT_AWS_SECRET_ACCESS_ID: admin
Error: YAML parse error on studio/templates/secret-studio.yaml: error converting YAML to JSON: yaml: line 22: could not find expected ':'
helm.go:84: [debug] error converting YAML to JSON: yaml: line 22: could not find expected ':'
```

Reply:
sorry I don't completely understand your last comment, but seems like you still have some indentation issue that I didn't experience in providing the value

Reply:
talked offline, convinced